tests/mcaptcha/default.nix

```diff
@@ -0,0 +1,103 @@
+{modules, ...}: let
+  port = 7000;
+  urlRoot = "http://localhost:${builtins.toString port}";
+in {
+  name = "mcaptcha tests";
+
+  nodes.create_locally = {pkgs, ...}: {
+    imports = [
+      modules.default
+      modules.mcaptcha
+    ];
+
+    services.mcaptcha.enable = true;
+    services.mcaptcha.settings.server.port = port;
+    services.mcaptcha.server.cookieSecretFile = pkgs.writeText "cookie-secret" "mcaptcha-cookie-secret-dm0tdGVzdC1ydW4tbWNhcHRjaGEtdGVzdHM";
+    services.mcaptcha.captcha.saltFile = pkgs.writeText "salt" "asdl;kjfhjawehfpa;osdkjasdvjaksndfpoanjdfainsdfaijdsfajlkjdsaf;ajsdfweroire";
+    services.mcaptcha.database.createLocally = true;
+    services.mcaptcha.redis.createLocally = true;
+
+    # the following two lines are for debugging
+    networking.firewall.enable = false;
+    services.mcaptcha.settings.server.ip = "0.0.0.0";
+  };
+
+  nodes.bring_your_own_services = {
+    pkgs,
+    config,
+    ...
+  }: {
+    imports = [
+      modules.default
+      modules.mcaptcha
+    ];
+
+    services.mcaptcha.enable = true;
+    services.mcaptcha.settings.server.port = port;
+    services.mcaptcha.settings.server.domain = "localhost";
+    services.mcaptcha.server.cookieSecretFile = pkgs.writeText "cookie-secret" "mcaptcha-cookie-secret-dm0tdGVzdC1ydW4tbWNhcHRjaGEtdGVzdHM";
+    services.mcaptcha.captcha.saltFile = pkgs.writeText "salt" "asdl;kjfhjawehfpa;osdkjasdvjaksndfpoanjdfainsdfaijdsfajlkjdsaf;ajsdfweroire";
+
+    services.mcaptcha.settings.database.name = "my_mcaptcha";
+    services.mcaptcha.settings.database.username = "role_mcaptcha";
+    services.mcaptcha.settings.database.hostname = "my_own_services";
+    services.mcaptcha.settings.database.port = 5432;
+    services.mcaptcha.database.passwordFile = pkgs.writeText "db-password" "mcaptcha-db-secret";
+
+    services.mcaptcha.redis.passwordFile = pkgs.writeText "redis-secret" "(*&(*):ps@r}";
+    services.mcaptcha.redis.host = "my_own_services";
+  };
+
+  nodes.my_own_services = {pkgs, ...}: {
+    imports = [
+      modules.default
+    ];
+    networking.firewall.enable = false;
+    services.postgresql.enable = true;
+    services.postgresql.enableTCPIP = true;
+    services.postgresql.initialScript = pkgs.writeText "postgresql-init-script" ''
+      CREATE ROLE role_mcaptcha WITH LOGIN PASSWORD 'mcaptcha-db-secret';
+      CREATE DATABASE my_mcaptcha;
+      GRANT ALL PRIVILEGES ON DATABASE my_mcaptcha TO role_mcaptcha;
+    '';
+    services.postgresql.authentication = ''
+      #type  database  DBuser         auth-method
+      host   all       all 0.0.0.0/0  md5
+    '';
+    services.redis.servers.mcaptcha.enable = true;
+    services.redis.servers.mcaptcha.port = 6379;
+    services.redis.servers.mcaptcha.bind = null;
+    services.redis.servers.mcaptcha.extraParams = [
+      "--loadmodule"
+      "${pkgs.mcaptcha-cache}/lib/libcache.so"
+    ];
+    services.redis.servers.mcaptcha.requirePass = "(*&(*):ps@r}";
```

Comment:
Where's the counterpart for `"(*&(*):ps@r}"`? Does it not have to be configured on the mcaptcha side as well? Consider using a variable. Then it's easy to find all usages of it.